### PR TITLE
Transform downcast pickle

### PR DESCRIPTION
**Pull request by blowekamp**
The issue report in #1112 is further addressed in this patch, with the pickling of derived Transform classes held by the the Transform base class.

### CODE REVIEW

**Comment by blowekamp** [comment]
The restoring of a a pickled object as a different Python type than the initial object may not be the best behavior.

Additionally, more testing needs to be added with the new `Downcast` method.

**Review by zivy — APPROVED** [review]

File: Code/Common/include/sitkTransform.h [hunk 1]
```diff
@@ -258,6 +260,15 @@ class SITKCommon_EXPORT Transform
    */
   void MakeUnique( );
 
+  /** \brief Get the TransformEnum of the underlying Transform.
+   *
+   *  A SimpleITK Transform object can internally hold any ITK transform. This
```
Comment (dave3d) [inline Code/Common/include/sitkTransform.h:265]: 'This returns' -> 'This method returns'

File: Code/Common/include/sitkTransform.h [hunk 2]
```diff
@@ -258,6 +260,15 @@ class SITKCommon_EXPORT Transform
...
+   *  returns the TransformEnum representing the internal ITK transform. This
+   *  value may be used to identify which SimpleITK class the transfom can be
```
Comment (dave3d) [inline Code/Common/include/sitkTransform.h:267]: 'transform' (sp)

File: Code/Common/src/sitkTransform.cxx [hunk 3]
```diff
@@ -578,18 +578,24 @@ void Transform::InternalInitialization(itk::TransformBase *transform)
   visitor.transform = transform;
   visitor.that = this;
 
-  // The following list must have the children before their parents to
-  // cast to the most derived classes
-  typedef typelist::MakeTypeList<itk::IdentityTransform<double, 2>,
+  // The following list must have the *parent* classed before their children
+  // since the list is traversed end to the beginning, and t
```
Comment (dave3d) [inline Code/Common/src/sitkTransform.cxx:598]: Did something get cut off at the end here?

File: Code/Common/include/sitkTransform.h (hunk 2 above)
Comment (blowekamp) [reply Code/Common/include/sitkTransform.h]: done

File: Code/Common/include/sitkTransform.h (hunk 1 above)
Comment (blowekamp) [reply Code/Common/include/sitkTransform.h:265]: done

**Review by dave3d — APPROVED** [review]

**Pull request merged**